app/models/test.rb

```diff
@@ -1,2 +1,8 @@
 class Test < ApplicationRecord
+  def self.by_category(name)
+    joins('JOIN categories on categories.id = tests.category_id')
+      .where('categories.title = ?', name)
+      .order('tests.title': :desc)
```

Comment:
```suggestion
      .order(title: :desc)
```
в принципе можно таблицу явно не указывать, если это обращение к дефолтной таблице